Teacher change schedule exception (WIP)

flexik opened the pull request:
This PR implements a new type of schedule exception for rewriting teachers assigned to events.

**TODO:**

- [ ] Load and assign students to affected events
- [ ] Add and test GIN index for `applied_schedule_exception_ids` event column
- [ ] Write exception docs
- [ ] Write specs

flexik commented:
This should be pretty much finished. Please take a look. @jirutka, @jnv

jirutka commented:
It looks good, but I miss specs for added behaviour in `assign_people`.

btw I’ve refactored `assign_people` a little bit. I’ve noticed missing specs after I had done it, so I already amended it instead of separate commit.

flexik commented:
I've changed how the exception works a bit. I am now parsing username list explicitly instead of relying on Postgres to implicitly convert username list string to array during UPDATE. This had to be done to properly detect whether the event record should be updated or not. Otherwise `updated_at` timestamp would be updated every time the import is ran.

Also the specs have been extended to cover this change and added behavior in `assign_people`.
